src/tools/rbd/action/Status.cc

```diff
@@ -190,7 +197,57 @@ static int do_show_status(librados::IoCtx& io_ctx, const std::string &image_name
     }
 
     if (!image_cache_str.empty()) {
+      json_spirit::mValue cache_state_json_root;
+      if(!json_spirit::read(image_cache_str.c_str(), cache_state_json_root)) {
+        std::cerr << "failed to parse image cache state." << std::endl;
         std::cout << "Image cache state: " << image_cache_str << std::endl;
+      } else {
+        auto cache_state = cache_state_json_root.get_obj();
+        try {
+          auto cache_type = cache_state["cache_type"].get_str();
+          auto update_time = cache_state["update_time"].get_str();
+          auto present = cache_state["present"].get_str();
```

Comment:
I found an issue here. I discussed with @tchaikov , he point out `encode_json()` will change `bool` to `string`. It might be more appropriate to use `f->dump_bool(`) . But this change involves a lot of code. In order to merge this pr faster, I still decode `bool `as `string`. I'll create a tracker and try to fix it later.
https://github.com/ceph/ceph/blob/bd85f2c1c5bd4073566c8806788da7c4e9501a0c/src/common/ceph_json.cc#L532

Reply:
Or, even better, consider switching to `json_spirit` in the cache plugin as well.